parser/body.go

```diff
@@ -40,76 +42,59 @@ func ParseBody(body []byte, board string, thread uint64, id uint64, ip string, i
 		}
 	}
 
-	start := 0
-	lineStart := 0
-	pyu := config.GetBoardConfigs(board).Pyu
+	bodyStr := strings.Trim(string(body), "\n")
+	boardConfig := config.GetBoardConfigs(board)
 
 	// Prevent link duplication
 	haveLink := make(map[uint64]bool)
 	// Prevent #pyu duplication
 	isSlut := false
+	// Prevent many cinema pushes in one post
+	pushedAlready := false
 
-	for i, b := range body {
-		switch b {
-		case '\n', ' ', '\t':
-		default:
-			if i == len(body)-1 {
-				i++
-			} else {
-				continue
+	for _, line := range strings.Split(bodyStr, "\n") {
+		if boardConfig.CinemaEnabled {
+			if match := common.CinemaPushCommandRegexp.FindStringSubmatch(line) ;
```

Comment:
Short circuit with `pushedAlready` first.  No need to run regex matching, if we have pushed one already.